universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java

```diff
@@ -126,14 +126,37 @@ private void putNull(Path path) {
    * Puts the given value at the given path in the underlying JSON data. Builds up the necessary
    * structure along the way, i.e., creates parent objects where necessary.
    *
+   * <p>If the path ends in an array (i.e. we are trying to add an element to a JSON array), this
+   * will check to make sure the array is there, then add the given element to the end of the array.
+   *
    * @param path the {@link Path} with the fully specified path, e.g.,
    *     "applicant.children[3].favorite_color.text" or the equivalent
    *     "$.applicant.children[3].favorite_color.text".
    * @param value the value to place; values of type Map will create the equivalent JSON structure
    */
   private void put(Path path, Object value) {
     putParentIfMissing(path);
-    putAt(path, value);
+    if (path.isArrayElement()) {
+      addArrayIfMissing(path);
+      addAt(path, value);
+    } else {
+      putAt(path, value);
+    }
+  }
+
+  /**
+   * Adds a JSON array at the given path, if it is not there already.
+   *
+   * @param path the path to the new array - must end with the array suffix [] or [index]
+   */
+  private void addArrayIfMissing(Path path) {
+    if (!path.isArrayElement()) {
```

Comment:
(blocking): this method shouldn't check here. The `path` argument should be used as-is, and it is the caller's responsibility to pass it the right string. So in `put` it should use `putArrayIfMissing(path.withoutArrayReference())`.

This way this method is dumb and just puts lists if missing, to whatever path is passed in.

Reply:
Got it! Updated